cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaBrokerConfigurationDiff.java

```diff
@@ -0,0 +1,201 @@
+            throw new RuntimeException("Failed to get broker " + brokerId + " configuration");
+        }
+        this.currentEntries = brokerConfigs.entries();
+        this.diff = computeDiff();
+    }
+
+    private KafkaConfiguration emptyKafkaConf() {
+        Map<String, Object> conf = new HashMap<>();
+        return new KafkaConfiguration(conf.entrySet());
+    }
+
+    private HashMap<String, String> configMap2Map() {
+        HashMap<String, String> result = new HashMap<String, String>();
+        if (desired == null || desired.getData() == null || desired.getData().get("server.config") == null) {
+            return result;
+        }
+        String desireConf = desired.getData().get("server.config");
+
+        List<String> list = getLinesWithoutCommentsAndEmptyLines(desireConf);
+        for (String line: list) {
+            String[] split = line.split("=");
+            if (split.length == 1) {
+                result.put(split[0], "");
+            } else {
+                result.put(split[0], split[1]);
+            }
+        }
+        return result;
+    }
+
+    private List<String> getLinesWithoutCommentsAndEmptyLines(String config) {
+        List<String> allLines = Arrays.asList(config.split("\\r?\\n"));
+        List<String> validLines = new ArrayList<>();
+
+        for (String line : allLines)    {
+            if (!line.replace(" ", "").startsWith("#") && !line.isEmpty())   {
+                validLines.add(line.replace(" ", ""));
+            }
+        }
+
+        return validLines;
+    }
+
+    private void fillPlaceholderValue(Map<String, String> map, String placeholder, String value) {
+        map.entrySet().forEach(entry -> {
+            if (!IGNORABLE_PROPERTIES.matcher(entry.getKey()).matches() && entry.getValue().contains("${" + placeholder + "}")) {
+                entry.setValue(entry.getValue().replaceAll("\\$\\{" + placeholder + "\\}", value));
+            }
+        });
+    }
+
+    private KafkaConfiguration computeDiff() {
+        HashMap<String, String> desiredMap = configMap2Map();
+        Map<String, String> currentMap = new HashMap<>();
+
+        currentEntries.stream().forEach(e -> {
+            currentMap.put(e.name(), e.value());
+        });
+
+        fillPlaceholderValue(desiredMap, "STRIMZI_BROKER_ID", Integer.toString(brokerId));
+
+        Map<String, String> diff = getMapDiff(currentMap, desiredMap);
+
+        String diffString = diff.toString();
+        diffString = diffString.substring(1, diffString.length() - 1).replace(", ", "\n");
+
+        return KafkaConfiguration.unvalidated(diffString);
+    }
+
+    public KafkaConfiguration getDiff() {
+        return this.diff;
+    }
+
+    @SuppressWarnings("checkstyle:Regexp")
+    private Map<String, String> getMapDiff(Map<String, String> current, Map<String, String> desired) {
+        Map<String, String> difference = new HashMap<>();
+        difference.putAll(desired);
+        desired.forEach((k, v) -> {
```

Comment:
There's no benefit here in using a `forEach`. Lambdas do not come for free, so try to use regular `for` loops when you aren't doing anything fancy.